client/src/pages/CalendarHomePage/CalendarHomePage.js

```diff
@@ -33,6 +36,13 @@ class CalendarHomePage extends Component {
     };
   }
 
+  showNewShowModal = show => {
```

Comment:
@kmid5280 is the `show` property necessary here? I don't see it being passed in from where `showNewShowModal` is called.

Reply:
@seankwilliams In the initialValues in Form.js, it checks to see if there's an existing show to determine whether to populate the date fields. If I remove the show param from the showNewShowModal function, then that ceases to work and it ends up populating the fields with the current date. The choice I think would be to either leave it as is or find a different way to show an empty field if the add button is clicked rather than a slot on the calendar.

Reply:
@kmid5280 Okay, I see. Right now, `show` in this function is set to a `MouseEvent` because it's firing from a click event. So while it's working, we'll want to change this a bit so it's not setting a bunch of values for `MouseEvent` in the `show.selected` state when that is expecting to receive a show. Otherwise, someone might try to rely on the data in `show.selected` in the future only to find that it's in an unexpected format.

To do this, rather than calling `selectShow`, you'll want to add a new Redux action that clears the selected show and sets the `show.selected` value to `{}`. And, I don't think it's necessary to pass `show` into `setModalVisibility` but I didn't go through the whole modal thoroughly so I'm not positive. If it is necessary, you'll want to pass in something like `{}` instead so it's passing in an empty show rather than a `MouseEvent`.

Let me know if that makes sense!

Reply:
> 
> 
> @kmid5280 Okay, I see. Right now, `show` in this function is set to a `MouseEvent` because it's firing from a click event. So while it's working, we'll want to change this a bit so it's not setting a bunch of values for `MouseEvent` in the `show.selected` state when that is expecting to receive a show. Otherwise, someone might try to rely on the data in `show.selected` in the future only to find that it's in an unexpected format.
> 
> To do this, rather than calling `selectShow`, you'll want to add a new Redux action that clears the selected show and sets the `show.selected` value to `{}`. And, I don't think it's necessary to pass `show` into `setModalVisibility` but I didn't go through the whole modal thoroughly so I'm not positive. If it is necessary, you'll want to pass in something like `{}` instead so it's passing in an empty show rather than a `MouseEvent`.
> 
> Let me know if that makes sense!

You're suggesting to implement a new action that clears the selected show. However by clicking the Add button, we are assuming that there is no show data to begin with. Is implementing this action supposed to be a defensive way of making sure there is no existing show data, or are we always going to have show data no matter the situation?

Reply:
@seankwilliams

Reply:
@kmid5280 It's a defensive way to be sure that if there is show data, it gets cleared. There will probably be some instances where show data isn't cleared from Redux so this will be sure the show data gets cleared when that happens.